src/dp_vnf.c

```diff
@@ -115,25 +181,16 @@ static __rte_always_inline bool dp_vnf_match(const struct dp_vnf *vnf, enum dp_v
 
 bool dp_vnf_lbprefix_exists(uint16_t port_id, uint32_t vni, struct dp_ip_address *prefix_ip, uint8_t prefix_len)
 {
-	struct dp_vnf *vnf;
-	uint32_t iter = 0;
-	int32_t ret;
-	const void *key;
+	struct dp_vnf vnf = {0};
 
-	while ((ret = rte_hash_iterate(vnf_handle_tbl, &key, (void **)&vnf, &iter)) != -ENOENT) {
-		if (DP_FAILED(ret)) {
-			if (prefix_ip->ip_type == RTE_ETHER_TYPE_IPV4)
-				DPS_LOG_WARNING("Iterating VNF table failed", DP_LOG_RET(ret), DP_LOG_PORTID(port_id),
-								DP_LOG_VNI(vni), DP_LOG_IPV4(prefix_ip->ipv4), DP_LOG_PREFLEN(prefix_len));
-			else
-				DPS_LOG_WARNING("Iterating VNF table failed", DP_LOG_RET(ret), DP_LOG_PORTID(port_id),
-								DP_LOG_VNI(vni), DP_LOG_IPV6(prefix_ip->ipv6), DP_LOG_PREFLEN(prefix_len));
-			return false;
-		}
-		if (dp_vnf_match(vnf, DP_VNF_TYPE_LB_ALIAS_PFX, port_id, vni, prefix_ip, prefix_len))
-			return true;
-	}
-	return false;
+	vnf.port_id = port_id;
+	vnf.vni = vni;
+	vnf.type = DP_VNF_TYPE_LB_ALIAS_PFX;
+	vnf.alias_pfx.length = prefix_len;
+	vnf.alias_pfx.ol.ip_type = prefix_ip->ip_type;
+	memcpy(&vnf.alias_pfx.ol, prefix_ip, sizeof(struct dp_ip_address));
```

Comment:
You should take advantage of the initialization code to prevent unneeded zeroing and then re-setting:
```
struct dp_vnf vnf = {
    .port_id = port_id,
    ... etc.
}
```
I *think* (but with relative confidence) that even `    .alias_pfx.ol = prefix_ip,` should work jsut fine as it is a shallow copy (even whough of an array).

Reply:
yeah, I changed to use the init form. and it works by judging from the test results.